src/mixins/java/org/spongepowered/common/mixin/core/world/entity/EntityMixin.java

```diff
@@ -247,70 +267,53 @@
-                            location.position(), location.position());
-                    if (SpongeCommon.post(event)) {
-                        return false;
-                    }
-                    destination = event.destinationPosition();
-                } else {
-                    destination = location.position();
+                destinationWorld = (ServerLevel) this.level;
+                destinationPosition = this.impl$fireMoveEvent(PhaseTracker.SERVER, location.position());
+                if (destinationPosition == null) {
+                    return false;
                 }
-                this.shadow$setPos(destination.x(), destination.y(), destination.z());
             }
 
-            if (!destinationWorld.getChunkSource().hasChunk((int) this.shadow$getX() >> 4, (int) this.shadow$getZ() >> 4)) {
-                // Roll back the position
-                this.shadow$setPos(originalPosition.x, originalPosition.y, originalPosition.z);
-                return false;
+            // If we have a player, there is different handling involved.
+            if (this instanceof ServerPlayerBridge) {
+                return ((ServerPlayerBridge) this).bridge$setPlayerLocation(originalDestinationWorld, destinationWorld, destinationPosition);
             }
 
             ((Entity) (Object) this).unRide();
+            if (isChangeOfWorld) {
+                final net.minecraft.server.level.ServerLevel originalWorld = (net.minecraft.server.level.ServerLevel) this.shadow$getCommandSenderWorld();
+                ((PlatformServerLevelBridge) this.shadow$getCommandSenderWorld()).bridge$removeEntity((Entity) (Object) this, true);
+                this.bridge$revive();
+                this.shadow$setLevel(destinationWorld);
+                destinationWorld.addFromAnotherDimension((Entity) (Object) this);
+
+                originalWorld.resetEmptyTime();
+                destinationWorld.resetEmptyTime();
+            }
 
-            final net.minecraft.server.level.ServerLevel originalWorld = (net.minecraft.server.level.ServerLevel) this.shadow$getCommandSenderWorld();
-            ((PlatformServerLevelBridge) this.shadow$getCommandSenderWorld()).bridge$removeEntity((Entity) (Object) this, true);
-            this.bridge$revive();
-            this.shadow$setLevel(destinationWorld);
-            destinationWorld.addFromAnotherDimension((Entity) (Object) this);
-
-            originalWorld.resetEmptyTime();
-            destinationWorld.resetEmptyTime();
-
-            final ChunkPos chunkPos = new ChunkPos((int) this.shadow$getX() >> 4, (int) this.shadow$getZ() >> 4);
-            destinationWorld.getChunkSource().addRegionTicket(TicketType.POST_TELEPORT, chunkPos, 1, ((Entity) (Object) this).getId());
+            return this.impl$teleportToWithTicket(destinationPosition.x(), destinationPosition.y(), destinationPosition.z(), false);
```

Comment:
Here is a good discussion.

Does setting the location of a non player trigger a chunk ticket? Should we be doing that?

Reply:
If the chunk isn't loaded and the entity is moved there, where would the entity be saved to? Does the chunk need to be loaded to save the entity's location to it?

Reply:
That is a good question dual. In fact, it is likely premature to discuss this now since entities still exist in Chunk files in 1.16.5. In 1.17, they exist outside chunk files and it likely makes a lot of sense to reposition them and do a save without triggering a chunk load.

Reply:
That's what I thought, I knew a split happened in 1.17.x - so I suspect that the answer is we need the ticket in 1.16, revisit in 1.17...